### PR TITLE
Implemented retention index computation according to Kovats/Van den Dool

### LINKED ISSUE #9: Implement Kovats retention index computation class
Implement the class computing the Kovats retention index based on reference alkanes.

### PR DESCRIPTION
**Pull request by hechth**
Implemented class for Kovats RI computation and added tests.

- Basic class
- Tests
- Extensions on data classes

Fixes #9

### CODE REVIEW

File: RIAssigner/compute/Kovats.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+
+class Kovats(ComputationMethod):
+    """ Class to compute the Kovats retention index. """
+
+    def compute(self, query: Data, reference: Data) -> List[float]:
+        """ Compute non-isothermal Kovats retention index.
+        For details see https://webbook.nist.gov/chemistry/gc-ri/.
+
+        Parameters
```
Comment (xtrojak) [inline RIAssigner/compute/Kovats.py:13]: Missing parameter `reference` in doc string.

File: RIAssigner/compute/Kovats.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+        return retention_indices
+
+
+def _get_bound_indices(target_rt: float, reference_rts: Iterable[Data.RetentionTimeType], lower_index: int, higher_index: int):
+    """ Get the indices of previosly eluting and next eluting reference compounds.
+    Retention times in 'Data' objects are sorted in ascending order, so this method assumes
+    that 'reference_rt' is sorted in ascending order.
+
+    Parameters
```
Comment (xtrojak) [inline RIAssigner/compute/Kovats.py:47]: Missing parameters in doc string.

File: RIAssigner/compute/Kovats.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    if target_rt > max(reference_rts) or higher_index >= len(reference_rts):
+        higher_index = len(reference_rts) - 1
+    else:
+        while reference_rts[higher_index] < target_rt:
+            higher_index += 1
+    lower_index = max(lower_index, higher_index - 1)
+    return lower_index, higher_index
+
+
+def _compute_ri(
```
Comment (xtrojak) [inline RIAssigner/compute/Kovats.py:62]: Missing doc string.

File: RIAssigner/data/MatchMSData.py [hunk 4]
```diff
@@ -91,7 +91,9 @@ def _spectrum_has_rt(spectrum: Spectrum) -> bool:
     has_key = 'retentiontime' in spectrum.metadata.keys()
```
Comment (xtrojak) [inline RIAssigner/data/MatchMSData.py:91]: You are basically checking the presence of `'retentiontime'` twice. Could be simplified by only using `safe_read_key` function.

File: RIAssigner/compute/Kovats.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+        lower_index = 0
+        higher_index = 0
+        retention_indices = []
+
+        # Copy rts and ris and insert 0 in the beginning, so that interpolation always starts at 0,0 to the first reference compound.
+        reference_rts = list(reference.retention_times)
+        reference_ris = list(reference.retention_indices)
+
+        reference_rts.insert(0, 0.0)
```
Comment (xtrojak) [inline RIAssigner/compute/Kovats.py:29]: Can be potentially simplified to 

```python
reference_rts = [0.0] + list(reference.retention_times)
```

**Review by xtrojak — COMMENTED** [review]
I think this looks nice, just some minor details like missing docs.

File: RIAssigner/compute/Kovats.py (hunk 5 above)
Comment (hechth) [reply RIAssigner/compute/Kovats.py:29]: Doesn't this cause 0 to be added to every element? I'll try and see if it works. Thanks for the hint, didn't know that.

Comment (martenson) [reply RIAssigner/compute/Kovats.py:29]:
```
>>> ['a','b']+[['c']]+[0.0]
['a', 'b', ['c'], 0.0]
```

File: RIAssigner/compute/Kovats.py (hunk 3 above)
Comment (hechth) [reply RIAssigner/compute/Kovats.py]: I added documentation to the original method that was renamed to _compute_kovats and the newer, high level method

File: RIAssigner/compute/Kovats.py (hunk 5 above)
Comment (hechth) [reply RIAssigner/compute/Kovats.py]: I included the proposed changes and tests are still all green, so seems good.

File: RIAssigner/data/MatchMSData.py (hunk 4 above)
Comment (hechth) [reply RIAssigner/data/MatchMSData.py]: I removed the function as it is no longer required.

File: RIAssigner/compute/Kovats.py (hunk 2 above)
Comment (hechth) [reply RIAssigner/compute/Kovats.py:64]: Docstring was added.

File: RIAssigner/compute/Kovats.py (hunk 1 above)
Comment (hechth) [reply RIAssigner/compute/Kovats.py:13]: Added documentation of compute function

**Review by xtrojak — APPROVED** [review]

**Review by VojtechBarton — APPROVED** [review]

**Review by maximskorik — APPROVED** [review]

**Pull request merged**